starter-core/src/main/java/io/micronaut/starter/build/maven/MavenScope.java

```diff
@@ -61,8 +62,10 @@ public static Optional<MavenScope> of(@NonNull Scope scope) {
                 if (scope.getPhases().contains(Phase.COMPILATION)) {
                     return Optional.of(MavenScope.PROVIDED);
                 }
+                if (scope.getPhases().contains(Phase.DEVELOPMENT)) {
+                    return Optional.of(MavenScope.COMPILE);
```

Comment:
If in the future we want to support this custom scope in the Maven Plugin, this mapping here will have to be removed (or return empty)

Reply:
the mapping is incorrect imo. `developmentOnly` should map to `provided` scope in Maven

Reply:
`provided` is probably closer, but still incorrect, because they still won't be available at "development runtime"

Reply:
you mean for `mn:run`? Is that because we aren't resolving provided dependencies in the Maven plugin?

`compile` is definitely wrong, it means it will be included in the production JAR

Reply:
If they are not used in Maven, then `PROVIDED` it should be. If we use `compile` they would end up in the distribution, which is not wanted (and has the potential to break things with native image since we include native libraries for file watching).

Reply:
resolved via https://github.com/micronaut-projects/micronaut-starter/pull/1215/commits/a22938647c55da74f3c4b3acbf795feae55f7e09